addons/account/models/account_move_line.py

```diff
@@ -1620,15 +1620,15 @@ def _search_panel_domain_image(self, field_name, domain, set_count=False, limit=
         if not isinstance(query, Query):
             return {}
 
-        query.order = None
-        query.add_where('account.id = account_move_line.account_id')
-        query_str, query_param = query.select()
-        self.env.cr.execute(f"""
+        self.env.cr.execute(
+            f"""
             SELECT account.root_id
-              FROM account_account account,
-                   LATERAL ({query_str}) line
+              FROM "account_move_line"
```

Comment:
You need to use the `from_clause` of the query, you might need to join other tables because of the domain